sky/spot/controller.py

```diff
@@ -140,11 +134,20 @@ def _run(self):
                         failure_type=spot_state.SpotStatus.FAILED,
                         end_time=end_time)
                     break
+                else:
+                    # Unexpected job status received, try to recover
+                    # the cluster to be safe.
+                    logger.info(f'Got unexpected job status: {job_status}. '
+                                'Recovering...')
+
+            # Failed to connect to the cluster or the cluster is partially down.
             # cluster can be down, INIT or STOPPED, based on the interruption
             # behavior of the cloud.
-            # Failed to connect to the cluster or the cluster is partially down.
-            # job_status is None or job_status == job_lib.JobStatus.FAILED
-            logger.info('The cluster is preempted.')
+            if is_preempted:
+                cluster_status_str = ('' if cluster_status is None else
+                                      f' (status: {cluster_status.value})')
+                logger.info(
+                    f'Cluster is preempted{cluster_status_str}. Recovering...')
```

Comment:
Let's move this part to above (see previous comment). We can remove the comment starting from L143.
```suggestion
            # Try to recover the spot jobs, when the cluster is preempted
            # or the job status is failed to be fetched.
```

Reply:
Fixed!